### ISSUE TITLE
Named serialization of inner values

### ISSUE THREAD

**Issue by spirali** [3Hren/msgpack-rust#217]
If there a posibility to serialize a structure into dictionary when it is inside an array?
It seems that `to_vec_named` works only for a top level structure.

```rust

#[derive(Debug, PartialEq, Serialize)]
struct Data {
    x: u32,
    y: u32
}
let data = Data {x: 123, y: 20}; 
rmps::encode::to_vec_named(&data).unwrap() // {'x': 123, 'y': 20}
rmps::encode::to_vec_named(&[data]).unwrap()); // [[123, 20]] 
```

**Comment by daboross** [comment]
What version of msgpack are you using?

This was broken on master branch for a while, but it should have been fixed as of #207 (see issue #205).

**Comment by spirali** [comment]
Thank you for information
